client/components/CompletedGraph.vue

```diff
@@ -1,93 +1,108 @@
+        var totalCount = 0; 
         if (this.statshistory && this.stats){
           this.datacollection = {};
           this.datacollection["labels"]= this.statshistory.map(eachStatistic => "" );
           this.datacollection["datasets"]=[];
           for (var thisPath in this.stats.FunctionStatsMap){
-            var dataSetForPath = {
-              label: thisPath + ": "+ this.stats.FunctionStatsMap[thisPath].Complete,
-              fill: true,                                       // Set to true because chart is stacked
-              backgroundColor: getBackgroundColorFor(thisPath), // Set color because chart is stacked
-              borderColor: getBorderColorFor(thisPath),
-              borderWidth: lineWidthInPixels,
-              radius:pointRadiusInPixels,
-              data: this.statshistory.map(eachStatistic => eachStatistic.FunctionStatsMap[thisPath].Complete)
-            };
-            this.datacollection["datasets"].push(dataSetForPath);
-             
+            if (this.routes==null || isPathIn(thisPath,this.routes)){
+              totalCount = totalCount + this.stats.FunctionStatsMap[thisPath].Complete;
+              var dataSetForPath = {
+                label: thisPath + ": "+ this.stats.FunctionStatsMap[thisPath].Complete,
+                fill: true,                                       // Set to true because chart is stacked
+                backgroundColor: getBackgroundColorFor(thisPath), // Set color because chart is stacked
+                borderColor: getBorderColorFor(thisPath),
+                borderWidth: lineWidthInPixels,
+                radius:pointRadiusInPixels,
+                data: this.statshistory.map(eachStatistic => eachStatistic.FunctionStatsMap[thisPath].Complete)
+              };
+              this.datacollection["datasets"].push(dataSetForPath);
+            }
           }
-          var legs = document.getElementById("completedGraphLegend");
-        
+          this.total = totalCount;
+          
           // legend  
+          var legs = document.getElementById("completedGraphLegend");
           var text = [];
           text.push('<ul class=\'' + 'chartLegend\'>');
           var chartDataDatasets = this.datacollection["datasets"];
           var chartDataDatasetsLength = chartDataDatasets.length;
-		  for (var i = 0; i < chartDataDatasets.length; i++) {
-			text.push('<li><span class=\'chartLabelEmblem\' style=\'' +
-			  'background-color:' + chartDataDatasets[i].backgroundColor + '; ' +
-		      'border-color:' + chartDataDatasets[i].borderColor + ';' +
-			  '\'></span>');
-			if (chartDataDatasets[i].label) {
-			  text.push('<span class=\'chartLabelText\'>'+chartDataDatasets[i].label+'</span>');
-			}
-			text.push('</li>');
-		  }
-		  text.push('</ul>');
-          legs.innerHTML  = text.join(''); 
+          for (var i = 0; i < chartDataDatasets.length; i++) {
```

Comment:
This code is part of the new function that I have created in `graphUtilities.js` and which is used by all four graphs. So no duplication now.

Reply:
This bit of code is explicitly generating the graph legend as HTML because the built-in legend does not allow the degree of customisation we require (e.g. it doesn't allow it to be above the graph). 

Whilst the chart.js docs appear to expect you to set innerHTML, since we're using Vue I agree it would be nice to make this a separate component. However I'd like to defer this as it is essentially a refinement of existing code rather than being related to the current changes. AT least we now only have a single copy of it.